src/app/problem/editor/right/Visible.js

```diff
@@ -1,31 +1,37 @@
 import {EditorButton} from "../../commons/EditorButton";
 import Block from "./Block";
-import {useState} from "react";
-import {useProblemEditorContext} from "../ProblemEditorContext";
+import {useEffect, useState} from "react";
+import {ACTION_UPDATE_VISIBILITY, useProblemEditorContext} from "../context";
 import {problemService} from "../../../../services/services";
-import {useParams} from "react-router-dom";
-
 
 const Visible = () => {
-    const {problemId} = useParams();
-    const {markProblemsDirty} = useProblemEditorContext();
+    const {problem, dispatch} = useProblemEditorContext();
     const [isOnClicked, setIsOnClicked] = useState(undefined);
 
+    useEffect(() => {
+       if (problem) {
+           if (isOnClicked === undefined) {
+               setIsOnClicked(problem.visible);
+           }
+       }
+    }, [problem, isOnClicked, setIsOnClicked]);
+
     const onVisibleButtonClick = (isVisibleButtonClicked) => {
-        setIsOnClicked(isVisibleButtonClicked);
-        problemService.updateProblemVisible(problemId, isVisibleButtonClicked)
-            .then(() => {
-                console.log("The problem's visibility has been modified");
-                markProblemsDirty();
-            });
+        if (isVisibleButtonClicked !== isOnClicked) {
+            setIsOnClicked(isVisibleButtonClicked);
```

Comment:
`isVisibleButtonClicked` and `isOnClicked` should be more readable.
Please rename them.